syntax = "proto3";

service RaftNode {
    rpc ackHeartbeat (heartbeatMessage) returns (ackMessage);
    rpc logAppend (logMessage) returns (ackMessage);
}

message heartbeatMessage {
    string leader_name = 1;
    int32 leader_port = 2;
}

message logMessage {
    int32 index = 1;
    string key = 2;
    string value = 3;
    string instruction = 4;
}

message ackMessage {
    bool success = 1;
    string message = 2;
}